app/models/service_template.rb

```diff
@@ -377,6 +377,13 @@ def create_resource_actions(ae_endpoints)
     save!
   end
 
+  def provision_request(userid, options = {})
+    workflow = ResourceActionWorkflow.new({}, User.find_by!(:userid => userid),
+                                          provision_action, :target => self)
+    options.each { |key, value| workflow.set_value(key, value) }
+    workflow.submit_request
```

Comment:
@mkanoor If we want to allow the API to call this it should be broken up into two methods.  One to be called from the API that returns the workflow to match what the current code does: [lib/services/api/service_template_workflow.rb#L7](https://github.com/ManageIQ/manageiq/blob/bc7fd6bf0d7d1181bfde269bf8a93650b906f174/lib/services/api/service_template_workflow.rb#L7)

And another which can be called when we need to create and submit the workflow.  This would support the two use cases from the API as well as what we need for control actions and automate.

Current API callers:
https://github.com/ManageIQ/manageiq/blob/bc7fd6bf0d7d1181bfde269bf8a93650b906f174/app/controllers/api/subcollections/service_templates.rb#L32
https://github.com/ManageIQ/manageiq/blob/bc7fd6bf0d7d1181bfde269bf8a93650b906f174/app/controllers/api/service_orders_controller.rb#L84